// components/pricing/index.js
import React, {Component} from 'react'
import MyButton from 'components/utils/MyButton'
import {Zoom} from 'react-reveal'

export default class Pricing extends Component{

    state = {
        prices: [100, 150, 200],
        positions: ['Balcony', 'Medium', 'Star'],
        desc: [
            'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur eum exercitationem impedit ipsum odio optio quibusdam quo repellat vel voluptate',
            'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur eum exercitationem impedit ipsum odio optio quibusdam quo repellat vel voluptate',
            'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequatur eum exercitationem impedit ipsum odio optio quibusdam quo repellat vel voluptate',
        ],
        linkto: ['http://sales/b', 'http://sales/m', 'http://sales/s'],
        delay: [500, 0, 500]

    }

    showBoxes = () => (
        this.state.prices.map((box, i) => (
            <Zoom key={i} delay={this.state.delay[i]}>
                <div className='pricing_item'>

                    <div className='pricing_inner_wrapper'>

                        <div className='pricing_title'>
                            <span>${this.state.prices[i]}</span>
                            <span>{this.state.positions[i]}</span>
                        </div>

                        <div className='pricing_description'>
                            {this.state.desc[i]}
                        </div>

                        <div className='pricing_buttons'>
                            <MyButton
                                path={this.state.linkto[i]}
                                name='Purchase'
                                bgc='#ffa800'
                                cl='white'
                                size='small'
                                />
                        </div>

                    </div>

                </div>
            </Zoom>
        ))
    )


    render(){
        return(
            <div className='bck_black'>
                <div className='center_wrapper pricing_section'>

                    <h2>Pricing</h2>

                    <div className='pricing_wrapper'>
                        { this.showBoxes() }
                    </div>

                </div>
            </div>
        )
    }
}
