// components/featured/Carousel.js
import React from 'react'
import Slider from 'react-slick'
import image1 from 'images/slide_one.jpg'
import image2 from 'images/slide_two.jpg'
import image3 from 'images/slide_three.jpg'


const Carousel = (props) => {

    const settings = {
        dots: false,
        infinite: true,
        autoplay: true,
        speed: 250,
        arrows: false
    }


    return (
        <div
        className='carrousel_wrapper'
        style={{
            overflow: 'hidden',
            height: `${window.innerHeight}px`,
            background: 'purple'
        }}
        >

            <Slider {...settings}>

                <div>
                    <div
                        className='carrousel_image'
                        style={{
                            background:` url(${image1})`,
                            height: `${window.innerHeight}px`
                        }}
                    >
                    </div>
                </div>
                <div>
                    <div
                        className='carrousel_image'
                        style={{
                            background:` url(${image2})`,
                            height: `${window.innerHeight}px`
                        }}
                    >
                    </div>
                </div>
                <div>
                    <div
                        className='carrousel_image'
                        style={{
                            background:` url(${image3})`,
                            height: `${window.innerHeight}px`
                        }}
                    >
                    </div>
                </div>

            </Slider>

        </div>
    )
}

export default Carousel
