// components/header/index.js
import React, {useState, useEffect} from 'react'
import 'components/header/style.css'

import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import MenuIcon from "@material-ui/icons/Menu";
import IconButton from "@material-ui/core/IconButton";
import SideDrawer from 'components/header/sideDrawer'

const Header = () => {

    const [state, setState] = useState({
        drawerOpen: false,
        headerShow: false
    })

    useEffect(() => {
            window.addEventListener('scroll', handleScroll)
    }, [])

    const handleScroll = () => {
        if(window.scrollY > 0){
            setState({
                ...state,
                headerShow: true
            })
        } else {
            setState({
                ...state,
                headerShow: false
            })
        }
    }


    const toggleDrawer = value => {
        setState({
            ...state,
            drawerOpen: value
        })
    }
        return(
            <div>
                <AppBar
                    position='fixed'
                    style={{
                        backgroundColor: state.headerShow ? '#2f2f2f' : 'transparent',
                        boxShadow: 'none',
                        padding: '10 0',
                        transition: 'background .3s ease-in-out'
                    }}
                >

                <Toolbar>

                    <div className="header_logo">

                        <div className='font_righteous header_logo_venue'>Avatar 2</div>

                        <div className='header_logo_title'>James Cameron</div>

                    </div>

                    <IconButton
                    aria-label='Menu'
                    color='inherit'
                    onClick={() => toggleDrawer(true)}
                    >
                        <MenuIcon />
                    </IconButton>

                    <SideDrawer
                    open={state.drawerOpen}
                    onClose={value => toggleDrawer(value)}
                    >

                    </SideDrawer>


                </Toolbar>

                </AppBar>

            </div>
        )
}

export default Header
